src/lex/cpp.rs

```diff
@@ -1417,13 +1435,28 @@ a";
 #define a c
 a
 ";
-        assert_same(src, "c");
+        assert_err!(
+            src,
+            CppError::IncompatibleRedefinition(_),
+            "incompatible redfinition"
+        );
         let src = "
 #define a b
 #define a
 a
 ";
-        assert_same(src, "");
+        assert_err!(
+            src,
+            CppError::IncompatibleRedefinition(_),
+            "incompatible redefinition"
+        );
+
+        let src = "
+#define a b
+#define a b
+a
+";
+        assert_same(src, "b");
```

Comment:
Can you also add a couple tests for redefining a function macro?

Reply:
Look 9 lines down ;)